pkg/webhook/validation/uservalidation.go

```diff
@@ -158,7 +157,7 @@ func checkCRDGroup(group string) bool {
 }
 
 // ValidateMCIdentity returns admission allowed/denied based on the member cluster's identity.
-func ValidateMCIdentity(ctx context.Context, client client.Client, userInfo authenticationv1.UserInfo, resourceNamespacedName types.NamespacedName, resourceKind, mcName string) admission.Response {
+func ValidateMCIdentity(ctx context.Context, client client.Client, userInfo authenticationv1.UserInfo, resourceNamespacedName types.NamespacedName, resourceKind, subResource, mcName string) admission.Response {
```

Comment:
Noticed I could just pass request here started making changes and realized it will a large PR for this PR will handle the refactor in a follow up

Reply:
I agree that the whole webhook should be refactored. However, if the request itself is passed in then the validator should be part of the handler. User validator should be independent from the request.

Reply:
#525 handles first refactor to pass request instead, cleans up logs and also improves UTs. Will handle separation of concerns between Fleet resource handler and user validation in another PR